src/js/utils/crypto/index.ts

```diff
@@ -0,0 +1,137 @@
+  let binary = "";
+  let bytes = new Uint8Array(buffer);
+  for (var i = 0; i < bytes.byteLength; i++) {
+    binary += String.fromCharCode(bytes[i]);
+  }
+  return window.btoa(binary);
+}
+
+function getCryptoApi(): SubtleCrypto {
+  const cryptoLib = window.crypto || (window as any).msCrypto;
+  return cryptoLib.subtle || (cryptoLib as any).webkitSubtle;
+}
+
+function printableBase64(base64: string): string {
+  // > To represent the encapsulated text of a PEM message, the encoding function's output is
+  // > delimited into text lines (using local conventions), with each line except the last
+  // > containing exactly 64 printable characters and the final line containing 64 or fewer
+  // > printable characters.
+
+  let result = "";
+
+  while (base64.length > 0) {
+    result += base64.substring(0, 64) + "\n";
+    base64 = base64.substring(64);
+  }
+
+  return result;
+}
+
+function privateKeyToPem(privateKey: CryptoKey): Promise<string> {
+  return new Promise((resolve, reject) => {
+    getCryptoApi()
+      .exportKey("pkcs8", privateKey)
+      .then(
+        exportedPrivateKey => {
+          const b64 = arrayBufferToBase64(exportedPrivateKey);
+          resolve(
+            "-----BEGIN PRIVATE KEY-----\n" +
+              printableBase64(b64) +
+              "-----END PRIVATE KEY-----"
+          );
+        },
+        error => {
+          reject(error);
+        }
+      );
+  });
+}
+
+function publicKeyToPem(publicKey: CryptoKey): Promise<string> {
+  return new Promise((resolve, reject) => {
+    getCryptoApi()
+      .exportKey("spki", publicKey)
+      .then(
+        exportedPublicKey => {
+          const b64 = arrayBufferToBase64(exportedPublicKey);
+          resolve(
+            "-----BEGIN PUBLIC KEY-----\n" +
+              printableBase64(b64) +
+              "-----END PUBLIC KEY-----"
+          );
+        },
+        error => {
+          reject(error);
+        }
+      );
+  });
+}
+
+function generatePrintableRSAKeypair({
+  bits,
+  hash,
+  publicExponent,
+  paddingScheme
+}: {
+  bits: 2048 | 4096;
+  hash: "SHA-256" | "SHA-512";
+  publicExponent: Uint8Array;
+  paddingScheme: "RSA-OAEP";
+}): Promise<[string, string]> {
+  return new Promise((resolve, reject) => {
+    getCryptoApi()
+      .generateKey(
+        {
+          name: paddingScheme,
+          modulusLength: bits,
+          publicExponent,
+          hash: { name: hash }
+        },
+        true,
+        ["encrypt", "decrypt"]
+      )
+      .then(
+        (pair: CryptoKeyPair) => {
+          Promise.all<string, string>([
+            privateKeyToPem(pair.privateKey),
+            publicKeyToPem(pair.publicKey)
+          ]).then(resolve);
+        },
+        error => {
+          reject(error);
+        }
+      );
+  });
+}
+
+// Defaults for key generation
+const BITS = 2048;
+const HASH = "SHA-256";
+const PADDING_SCHEME = "RSA-OAEP";
+const PUBLIC_EXPONENT = new Uint8Array([0x01, 0x00, 0x01]);
```

Comment:
# 🚫 ❓ 

the cli uses `65537`, also the docs of the lib used by the cli claims that 

> Unless you have a specific reason to do otherwise, you should always use 65537.

see https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/?highlight=rsa#generation

do we have a specific reason?

Reply:
oh boy! i just realized that it's about bYtes! nevermind :)

Reply:
Yeah took me a while to decode this, too. I thought this was decimal 5